File: experiments/016-LanguageComparisonRepeated/cells/java-gradle_1-Empty_free/2019-2c-parcial2-masked/20260917-053838-37234/output/project/src/main/java/surveycrawler/Boulder.java
package surveycrawler;

public class Boulder extends GroundType {

    // moving

    @Override
    public void moveTowards(SurveyCrawler aCrawler, Point aDirection) {
        aCrawler.signalBoulderFound();
    }

    // facing

    @Override
    public void assertCanTurn(SurveyCrawler aCrawler) {
        aCrawler.signalBoulderFound();
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2019-2c-parcial2-masked/20260921-210220-97414/analysis/acceptance-project/src/surveycrawler/boulder.py
from surveycrawler.ground import Ground


class Boulder(Ground):

    # exceptions

    @classmethod
    def boulder_found_error_description(cls):
        return "Can not move nor turn on a boulder"

    def signal_boulder_found(self):
        raise RuntimeError(self.boulder_found_error_description())

    # moving

    def move_crawler_towards(self, a_crawler, a_direction):
        self.signal_boulder_found()

    # testing

    def assert_can_be_occupied(self):
        self.signal_boulder_found()
